[BUG] Spectrogram example does not compile

cc @mattpap 

`bokeh info` output:
```
(23test) ❯ bokeh info
Python version      :  3.8.5 (default, Sep  4 2020, 02:22:02)
IPython version     :  7.20.0
Tornado version     :  6.1
Bokeh version       :  2.3.0
BokehJS static path :  /Users/bryan/anaconda/envs/23test/lib/python3.8/site-packages/bokeh/server/static
node.js version     :  v15.10.0
npm version         :  7.5.3
```
log output
```
(23test) ❯ bokeh serve --show spectrogram
2021-02-24 10:20:26,225 Starting Bokeh server version 2.3.0 (running on Tornado 6.1)
2021-02-24 10:20:26,227 User authentication hooks NOT provided (default user enabled)
2021-02-24 10:20:26,230 Bokeh app running at: http://localhost:5006/spectrogram
2021-02-24 10:20:26,230 Starting Bokeh server with process id: 88180
Compilation failed:

spectrogram/waterfall.ts:9:3 - error TS2416: Property 'model' in type 'WaterfallRendererView' is not assignable to the same property in base type 'RendererView'.
  Type 'WaterfallRenderer' is not assignable to type 'Renderer'.
    Types of property '__view_type__' are incompatible.
      Type 'WaterfallRendererView' is not assignable to type 'RendererView'.
        Property 'canvas' is private in type 'WaterfallRendererView' but not in type 'RendererView'.

9   model: WaterfallRenderer
    ~~~~~
spectrogram/waterfall.ts:11:11 - error TS2416: Property 'canvas' in type 'WaterfallRendererView' is not assignable to the same property in base type 'RendererView'.
  Type 'HTMLCanvasElement[]' is missing the following properties from type 'CanvasView': model, bbox, underlays_el, primary, and 45 more.

11   private canvas: HTMLCanvasElement[]
             ~~~~~~
spectrogram/waterfall.ts:122:18 - error TS2430: Interface 'WaterfallRenderer' incorrectly extends interface 'Renderer'.
  Types of property '__view_type__' are incompatible.
    Type 'WaterfallRendererView' is not assignable to type 'RendererView'.

122 export interface WaterfallRenderer extends WaterfallRenderer.Attrs {}
                     ~~~~~~~~~~~~~~~~~
spectrogram/waterfall.ts:126:3 - error TS2416: Property '__view_type__' in type 'WaterfallRenderer' is not assignable to the same property in base type 'Renderer'.
  Type 'WaterfallRendererView' is not assignable to type 'RendererView'.

126   __view_type__: WaterfallRendererView
      ~~~~~~~~~~~~~
```

--- a/examples/app/spectrogram/audio.py
+++ b/examples/app/spectrogram/audio.py
@@ -1,7 +1,7 @@
 from time import sleep
 
 import numpy as np
-import scipy as sp
+from scipy.fft import fft
 from scipy.integrate import simps
 
 NUM_SAMPLES = 1024
@@ -30,8 +30,7 @@ try:
             try:
                 raw_data  = np.fromstring(stream.read(NUM_SAMPLES), dtype=np.int16)
                 signal = raw_data / 32768.0
-                fft = sp.fft(signal)
-                spectrum = abs(fft)[:int(NUM_SAMPLES/2)]
+                spectrum = abs(fft(signal))[:int(NUM_SAMPLES/2)]
                 power = spectrum**2
                 bins = simps(np.split(power, NUM_BINS))
                 data['values'] = signal, spectrum, bins
@@ -63,8 +62,7 @@ except ImportError:
             A = 0.4 + 0.05 * np.random.random()
             signal = A * fm_modulation(_t, _f_carrier, _f_mod, _ind_mod)
 
-            fft = sp.fft(signal)
-            spectrum = abs(fft)[:int(NUM_SAMPLES/2)]
+            spectrum = abs(fft(signal))[:int(NUM_SAMPLES/2)]
             power = spectrum**2
             bins = simps(np.split(power, NUM_BINS))
             data['values'] = signal, spectrum, bins

--- a/examples/app/spectrogram/waterfall.py
+++ b/examples/app/spectrogram/waterfall.py
@@ -8,7 +8,7 @@ class WaterfallRenderer(Renderer):
 
     __implementation__ = join(dirname(__file__), "waterfall.ts")
 
-    latest = Seq(Float)
+    latest = Seq(Float, default=[])
 
     palette = Seq(Color)
 

--- a/examples/app/spectrogram/waterfall.ts
+++ b/examples/app/spectrogram/waterfall.ts
@@ -8,8 +8,8 @@ import * as p from "core/properties"
 export class WaterfallRendererView extends RendererView {
   model: WaterfallRenderer
 
-  private canvas: HTMLCanvasElement[]
-  private image: Uint32Array[]
+  private canvases: HTMLCanvasElement[]
+  private images: Uint32Array[]
   private x: number[]
   private col: number
   private tile: number
@@ -24,11 +24,11 @@ export class WaterfallRendererView extends RendererView {
     const N = Math.ceil(this.model.num_grams/this.model.tile_width) + 1
     const [w, h] = [this.model.tile_width, this.model.gram_length]
 
-    this.canvas = []
-    this.image = []
+    this.canvases = []
+    this.images = []
     for (let i = 0; i < N; i++) {
-      this.canvas.push(canvas({width: w, height: h}))
-      this.image.push(new Uint32Array(w*h))
+      this.canvases.push(canvas({width: w, height: h}))
+      this.images.push(new Uint32Array(w*h))
     }
 
     this.x = new Array(N)
@@ -67,7 +67,7 @@ export class WaterfallRendererView extends RendererView {
     ctx.translate(0, -sy)
 
     for (let i = 0; i < sx.length; i++)
-      ctx.drawImage(this.canvas[i], sx[i], sy, sw, sh)
+      ctx.drawImage(this.canvases[i], sx[i], sy, sw, sh)
 
     ctx.translate(0, sy)
     ctx.scale(1, -1)
@@ -97,12 +97,12 @@ export class WaterfallRendererView extends RendererView {
     // apply the lastest column to the current tile image
     const buf32 = new Uint32Array(this.cmap.rgba_mapper.v_compute(this.model.latest).buffer)
     for (let i = 0; i < this.model.gram_length; i++)
-      this.image[this.tile][i*this.model.tile_width+this.col] = buf32[i]
+      this.images[this.tile][i*this.model.tile_width+this.col] = buf32[i]
 
     // update the tiles canvas with the image data
-    const cctx = this.canvas[this.tile].getContext('2d')!
+    const cctx = this.canvases[this.tile].getContext('2d')!
     const image = cctx.getImageData(0, 0, this.model.tile_width, this.model.gram_length)
-    image.data.set(new Uint8Array(this.image[this.tile].buffer))
+    image.data.set(new Uint8Array(this.images[this.tile].buffer))
     cctx.putImageData(image, 0, 0)
   }
 }
@@ -133,7 +133,7 @@ export class WaterfallRenderer extends Renderer {
     this.prototype.default_view = WaterfallRendererView
 
     this.define<WaterfallRenderer.Props>(({Int, Number, Color, Array}) => ({
-      latest:      [ Array(Number) ],
+      latest:      [ Array(Number), [] ],
       palette:     [ Array(Color) ],
       num_grams:   [ Int ],
       gram_length: [ Int ],

